### PR TITLE
fix: update gql error handler

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes 

### Additional details

Update error handling on gql cloud requests, so the errors can be propagated to the UI and we can handle them with some alerts

[Here](https://app.circleci.com/pipelines/github/cypress-io/cypress/38252/workflows/de5bc3d0-0345-425c-9538-14b7409cf991/jobs/1549246) is a failing test with the previous implementation - that is passing with this new change

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/data-context/src/sources/CloudDataSource.ts [hunk 1]
```diff
@@ -174,6 +153,22 @@ export class CloudDataSource {
...
 
+  #formatWithErrors (data: OperationResult<any, any>) {
+    // If we receive a 401 from the dashboard, we need to logout the user
+    if (data.error?.response?.status === 401) {
+      this.params.logout()
+    }
+
+    if (data.error?.graphQLErrors[0]) {
+      this.params.onError(data.error?.graphQLErrors[0])
+    }
```
Comment (tgriesser) [inline packages/data-context/src/sources/CloudDataSource.ts:164]: I think we can get rid of this and the `onError` we pass to `CloudDataSource` entirely

**Review by tgriesser — CHANGES_REQUESTED** [review]

File: packages/data-context/src/sources/CloudDataSource.ts (hunk 1 above)
Comment (estrada9166) [reply packages/data-context/src/sources/CloudDataSource.ts]: Updated in [cd05ee9](https://github.com/cypress-io/cypress/pull/21542/commits/cd05ee95576355255aa228cbac2f2ff5b58a434b)

File: packages/data-context/src/sources/CloudDataSource.ts [hunk 2]
```diff
@@ -174,6 +152,18 @@ export class CloudDataSource {
     return `${print(config.document)}-${stringifyVariables(config.variables)}`
   }
 
+  #formatWithErrors (data: OperationResult<any, any>) {
+    // If we receive a 401 from the dashboard, we need to logout the user
```
Comment (tgriesser) [inline packages/data-context/src/sources/CloudDataSource.ts:156]:
```suggestion
  #formatWithErrors: (data: OperationResult<any, any>) => {
    // If we receive a 401 from the dashboard, we need to logout the user
```

Needs to be an arrow fn to reference `this` correctly the way we're passing it to `then`

**Review by tgriesser — CHANGES_REQUESTED** [review]

File: packages/data-context/src/sources/CloudDataSource.ts [hunk 3]
```diff
@@ -183,6 +152,23 @@ export class CloudDataSource {
...
+      // Do nothing on failure here
+      this.invalidate({ __typename: 'Query' }).catch()
+    }
+
+    return {
+      ...data,
+      errors: data.error?.graphQLErrors,
+    }
+  }
+
```
Comment (tgriesser) [inline packages/data-context/src/sources/CloudDataSource.ts:171]:
```suggestion
  #formatWithErrors = async (data: OperationResult<any, any>) => {
    // If we receive a 401 from the dashboard, we need to logout the user
    if (data.error?.response?.status === 401) {
      await this.params.logout()
    }

    if (data.error && data.operation.kind === 'mutation') {
      await this.invalidate({ __typename: 'Query' })
    }

    return {
      ...data,
      errors: data.error?.graphQLErrors,
    }
  }

```

**Review by tgriesser — CHANGES_REQUESTED** [review]

**Review by tgriesser — APPROVED** [review]

**Pull request merged**